pkg/commands/workload_apply_test.go

```diff
@@ -7751,6 +7751,119 @@ To get status: "tanzu apps workload get spring-petclinic"
 			},
 			ShouldError: true,
 		},
+		{
+			Name: "update - replace without adding lsp annotation",
```

Comment:
is there a test using `--source-image` as mentioned in the bug?, if not can we create one

Reply:
Currently, we do not have a way to test `--source-image` in the `apply` tests, I can add a e2e test and fill an issue to create tests for this kind of integration.